pkg/ddevapp/config.go

```diff
@@ -212,22 +284,40 @@ func (c *Config) Hostname() string {
 
 // WriteDockerComposeConfig writes a docker-compose.yaml to the app configuration directory.
 func (c *Config) WriteDockerComposeConfig() error {
-	log.WithFields(log.Fields{
-		"Location": c.DockerComposeYAMLPath(),
-	}).Debug("Writing docker-compose.yaml")
+	var err error
 
-	f, err := os.Create(c.DockerComposeYAMLPath())
-	if err != nil {
-		return err
-	}
-	defer util.CheckClose(f)
+	if !fileutil.FileExists(c.GetPath("provider.env")) {
+		log.WithFields(log.Fields{
+			"Location": c.GetPath("provider.env"),
+		}).Debug("Writing blank provider.env")
 
-	rendered, err := c.RenderComposeYAML()
-	if err != nil {
-		return err
+		// Create an empty provider.env file to make docker-compose happy.
+		_, err := os.Create(c.GetPath("provider.env"))
```

Comment:
An alternative would be to write a docker-compose.pantheon.yaml that sets the environment variables (or adds an env_file directive). This could potentially be a file managed by ddev.